### PR TITLE
Disable coreOS auto-updates

### LINKED ISSUE #1240: [Proposal] Disable CoreOS update-engine
We're having some issues due to CoreOS update-engine. Some machines has been upgraded to a newer CoreOS version. Immutability is important and this is causing upgrades to docker `17.09.1-ce` that has serious issues like: #1135

### PR DESCRIPTION
**Pull request by jorge07**
Fixes #1240

### CODE REVIEW

**Comment by jorge07** [comment]
Not tested yet

File: core/controlplane/config/templates/cluster.yaml [hunk 1]
```diff
@@ -13,6 +13,9 @@ s3URI: {{.S3URI}}
 # The AMI ID of CoreOS.
 amiId: "{{.AmiId}}"
 
+# CoreOS has automatic updates https://coreos.com/os/docs/latest/update-strategies.html. This can be a risk in certain situations. Set this param to true to disable it in all instances.
+disableCoreOSUpdates: false
```
Comment (mumoshu) [inline core/controlplane/config/templates/cluster.yaml]: I'd prefer `disableContainerLinuxAutomaticUpdates` as it has officially renamed to Container Linux a year or so ago 😃

**Comment by jorge07** [comment]
@mumoshu I updated the PR renaming this. I did not have time to test it yet, but I'll try to do it asap.

**Comment by davidmccormick** [comment]
I think that, once tested, this should be on by default - not behind a feature gate. :)

**Comment by mumoshu** [comment]
@davidmccormick Ah! That's definitely a good point. I agree.

**Comment by jorge07** [comment]
Sorry for this waiting for a week. I've the remote team in the office and it's a super busy week. I hope find some time for rebase the branch and launch the cluster next week as I think it would be great have it ready for 0.9.10 final release

**Comment by mumoshu** [comment]
@jorge07 No worry but I just want to sync up with you that I am about to cut v0.9.10 this week! Do you have any reason you want this before #1233 which is I'm going to merge soon after cutting v0.9.10?

**Comment by jorge07** [comment]
I did the QA today and this is not working at all.

The `update-engine.service` is running event is `sysctemctl status` show it as masked.

If I restart the service it fails, obviously because is masked. Any suggestions?

**Comment by mumoshu** [comment]
@jorge07 Thanks for your efforts. Unfortunately, it seems like we need ignition to actually disable update-engine according to https://github.com/coreos/bugs/issues/1646 and linked issues...

An alternative idea is that to automatically remove `update-engine` related systemd unit file somewhere in the filesystem so that update-engine would never start in case of a reboot.

**Comment by mumoshu** [comment]
Hm, or maybe just specifying both `command: stop` and `mask: true` would do the job?

ref: https://github.com/coreos/bugs/issues/1982#issuecomment-304957249

**Comment by jorge07** [comment]
I tried masking the update engine with the stop option too. Same issue, it ends as `failed`. The cluster its created and it works, but having a failed unit its not nice. Should I start considering refactor into ignition? Sounds like a huge epic

**Comment by mumoshu** [comment]
@jorge07 Thanks for your efforts. If you still want to make it in v0.9.10, I'd recommend my suggestion of removing update-engine(and possibly locksmithd related units).

It would look something like introducing `disable-automatic-update.service` which would basically run:

```
for u in update-engine locksmithd; do
  find / | grep lib64/systemd | grep $u | xargs rm
  systemctl stop ${u}.service
  systemctl disable ${u}.service
fi
```

You probably need this to clear failed units:

```
systemctl daemon-reload
systemctl reset-failed
```

But beware that this is completely untested!

**Comment by mumoshu** [comment]
Of course, I'd appreciate it if you could work on the migration to ignition 👍

**Comment by jorge07** [comment]
I thought the same, I think that create a work around unit that can stop it and clean the failures it's the best option meanwhile migrate into ignition. I'll work on this next week. Thanks for the suggestions!

**Comment by mumoshu** [comment]
@jorge07 LGTM. Thank you very much for your effort 🍻

**Pull request merged**